src/mimeparser.rs

```diff
@@ -1972,6 +1972,19 @@ fn is_attachment_disposition(mail: &mailparse::ParsedMail<'_>) -> bool {
             .any(|(key, _value)| key.starts_with("filename"))
 }
 
+fn is_valid_deltachat_vcard(mail: &mailparse::ParsedMail) -> bool {
+    let Ok(body) = &mail.get_body() else {
+        return false;
+    };
+    let contacts = deltachat_contact_tools::parse_vcard(body);
+    if let Some(contact) = contacts.first() {
```

Comment:
It doesn't matter for now since we're concentrating on showing vCards sent by Delta Chat, but note that if we go for "any contact" instead of "first contact", we either need to tell the UI to search through the list for a contact with an email addr, or to filter out contacts without an email, in order to avoid crashes when someone sends a vCard with multiple contacts in it.

BTW, one final other possibility would be to just always return `false` if the vCard doesn't contain exactly one contact. Which I actually like, because then we don't have to worry about very big vCards with lots of contacts, which you brought up in the vcard-contact-sharing group.

Reply:
> BTW, one final other possibility would be to just always return `false` if the vCard doesn't contain exactly one contact. Which I actually like, because then we don't have to worry about very big vCards with lots of contacts,

+1

Reply:
Then my change here needs to be just reverted i guess